File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/Mark/test_read_binary_file.cc
#include "STMDAQ-TestBeam/utils/BinaryFile.hh"
#include <string> 
#include <sstream>                                                                     

int main(){  
	BinaryFile *bf = new BinaryFile();
	bf->open_input_file("data/stm_hzdr_raw__run0000088_subrun00000.bin");
	unsigned int filesize = bf->fsize();
	std::cout << "File: " << bf->filename() << " Filesize = " << filesize << " bytes " << std::endl;

//  Read the whole file
	unsigned int N = filesize/2;  // #int16 is filesize in bytes / 2 
	int16_t* data = new int16_t[N];
	bf->read_data(data,N); // read whole file
	for (unsigned int i = 0; i < 100; i++){
		std::cout << "data[" << i << "] = " << data[i] << std::endl;
	}

//  Read subset of file with a given offset (in bytes) from the start of the file
	unsigned int byte_offset_from_start = 2*512;
	N = 512; // # of int16 values to read  
	data = new int16_t[N];
	bf->read_data(data,N,byte_offset_from_start); // read subset of file
	for (unsigned int i = 0; i < 50; i++){
		std::cout << "data[" << i << "] = " << data[i] << std::endl;
	}

	bf->close_input_file();
	delete bf;
//
}

File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/Mark/test_read_binary_file.py
import sys
import os
import BinaryFile as bf
import xml.etree.ElementTree as XML

def test_xml():
	# test code to read xml file
	xml = {}
	STMDAQ_ROOT = os.getenv('STMDAQ_ROOT')
	xml_file = STMDAQ_ROOT+"/config/stmdaq.xml"
	xroot = XML.parse(xml_file).getroot()
	for value in xroot.iter():
		if (value.tag != 'stm') and ('fmc144' in value.tag):
		    xml[value.tag] = value.text.strip()

	print("Setting Register values using xml file = %s" % (xml_file))
	print(xml)
	for key in xml:
	    ivalue = int (xml[key], 16)
	    print(key)
	    print(" %s = 0x%x (%d)" % (key, ivalue, ivalue)) 

	print(int (xml["fmc144_ext_ADC_offset"], 16))
	print(int (xml["fmc144_Start_reset_stop_clock"],16))
	print(int (xml["fmc144_Test_beam_10g_readout"],16))
	print(int (xml["fmc144_ext_slice_length"],16))
	print(int (xml["fmc144_ext_slice_num"],16))
	print(int (xml["fmc144_ext_int_delay"],16))
	print(int (xml["fmc144_int_ADC_offset"],16))
	print(int (xml["fmc144_int_slice_length"],16))
	print(int (xml["fmc144_int_slice_num"],16))
	print(int (xml["fmc144_ext_trig_timeout"],16))



def debug(data):
	i = 0
	for d in data:
		print("[%d] = %d" % (i,d))
		i = i + 1

#test_xml()

bff = bf.BinaryFile("data/stm_hzdr_0000012_00026.bin")

data = bff.read_data() # read whole file
#debug(data)

read_from_byte_offset = 2*512
num_int16_to_read = 512
data = bff.read_data(read_from_byte_offset, num_int16_to_read) # read part of file
#debug(data)

#data = bff.read_data(read_from_byte_offset, 1000*num_int16_to_read) # check exits if attempt to read too many bytes
#data = bff.read_data(-1) # check rogue argument error



bff.close_file()
